apis/python/src/tiledbsoma/soma_object.cc

```diff
@@ -57,52 +57,41 @@ void load_soma_object(py::module& m) {
-                    else if (soma_obj_type == "somasparsendarray")
-                        return py::cast(
-                            dynamic_cast<SOMASparseNDArray&>(*soma_obj));
-                    else if (soma_obj_type == "somadensendarray")
-                        return py::cast(
-                            dynamic_cast<SOMADenseNDArray&>(*soma_obj));
-                    else if (soma_obj_type == "somacollection")
-                        return py::cast(
-                            dynamic_cast<SOMACollection&>(*soma_obj));
-                    else if (soma_obj_type == "somaexperiment")
-                        return py::cast(
-                            dynamic_cast<SOMAExperiment&>(*soma_obj));
-                    else if (soma_obj_type == "somameasurement")
-                        return py::cast(
-                            dynamic_cast<SOMAMeasurement&>(*soma_obj));
-                    else if (soma_obj_type == "somascene")
-                        return py::cast(dynamic_cast<SOMAScene&>(*soma_obj));
-                    else if (soma_obj_type == "somamultiscaleimage")
-                        return py::cast(
-                            dynamic_cast<SOMAMultiscaleImage&>(*soma_obj));
-                    return py::none();
-                } catch (...) {
-                    return py::none();
-                }
+                std::transform(
+                    soma_obj_type->begin(),
+                    soma_obj_type->end(),
+                    soma_obj_type->begin(),
+                    [](unsigned char c) { return std::tolower(c); });
+
+                if (soma_obj_type == "somadataframe")
+                    return py::cast(dynamic_cast<SOMADataFrame&>(*soma_obj));
+                if (soma_obj_type == "somapointclouddataframe")
+                    return py::cast(
+                        dynamic_cast<SOMAPointCloudDataFrame&>(*soma_obj));
+                else if (soma_obj_type == "somasparsendarray")
+                    return py::cast(
+                        dynamic_cast<SOMASparseNDArray&>(*soma_obj));
+                else if (soma_obj_type == "somadensendarray")
+                    return py::cast(dynamic_cast<SOMADenseNDArray&>(*soma_obj));
+                else if (soma_obj_type == "somacollection")
+                    return py::cast(dynamic_cast<SOMACollection&>(*soma_obj));
+                else if (soma_obj_type == "somaexperiment")
+                    return py::cast(dynamic_cast<SOMAExperiment&>(*soma_obj));
+                else if (soma_obj_type == "somameasurement")
+                    return py::cast(dynamic_cast<SOMAMeasurement&>(*soma_obj));
+                else if (soma_obj_type == "somascene")
+                    return py::cast(dynamic_cast<SOMAScene&>(*soma_obj));
+                else if (soma_obj_type == "somamultiscaleimage")
+                    return py::cast(
+                        dynamic_cast<SOMAMultiscaleImage&>(*soma_obj));
```

Comment:
This code removed the else clause that caught unknown soma types.  The C++ compiler generates an error suggesting this error:
```
  src/tiledbsoma/soma_object.cc: In lambda function:
  src/tiledbsoma/soma_object.cc:95:13: warning: control reaches end of non-void function [-Wreturn-type]
     95 |             },
        |             ^
```

I think we probably need a final `else` statement in this if/else chain.

CC: @nguyenv